school/redfish/computersystem.go

```diff
@@ -300,6 +294,55 @@ type Boot struct {
 	UefiTargetBootSourceOverride string
 }
 
+// UnmarshalJSON unmarshals a Boot object from the raw JSON.
+func (boot *Boot) UnmarshalJSON(b []byte) error {
+	type temp Boot
+	var t struct {
+		temp
+		BootOptions common.Link
+	}
+
+	err := json.Unmarshal(b, &t)
+	if err != nil {
+		return err
+	}
+
+	*boot = Boot(t.temp)
+
+	// Extract the links to other entities for later
+	boot.bootOptions = string(t.BootOptions)
+
+	return nil
+}
+
+// Reset describe the type off reset to be issue by the resource
+type ResetType string
+
+const (
+	// On shall be used to power on the machine
+	On ResetType = "On"
```

Comment:
Since there's some overlap in enum names, I've been following the convention of appending the type name to the name to make it clear and to prevent conflicts.

I think it would be good to keep following this convention, so that would make these OnResetType, ForceOffResetType, etc. A little more verbose, but at least in a couple cases it was necessary.